featuretools/primitives/standard/transform_primitive.py

```diff
@@ -906,3 +907,138 @@ def lag(time_index, numeric):
+    commutative = True
+
+    def get_function(self):
+        def geomidpoint_func(latlong_1, latlong_2):
+            lat_1s = np.array([x[0] if isinstance(x, tuple) else np.nan for x in latlong_1])
+            lon_1s = np.array([x[1] if isinstance(x, tuple) else np.nan for x in latlong_1])
+            lat_2s = np.array([x[0] if isinstance(x, tuple) else np.nan for x in latlong_2])
+            lon_2s = np.array([x[1] if isinstance(x, tuple) else np.nan for x in latlong_2])
+            lat_middle = np.array([lat_1s, lat_2s]).transpose().mean(axis=1)
+            lon_middle = np.array([lon_1s, lon_2s]).transpose().mean(axis=1)
+            return list(zip(lat_middle, lon_middle))
+        return geomidpoint_func
+
+
+class CityblockDistance(TransformPrimitive):
+    """Calculates the distance between points in a city road grid.
+
+    Description:
+        This distance is calculated using the haversine formula, which
+        takes into account the curvature of the Earth.
+        If either input data contains `NaN`s, the calculated
+        distance with be `NaN`.
+        This calculation is also known as the Mahnattan distance.
+
+    Args:
+        unit (str): Determines the unit value to output. Could
+            be miles or kilometers. Default is miles.
+
+    Examples:
+        >>> cityblock_distance = CityblockDistance()
+        >>> DC = (38, -77)
+        >>> Boston = (43, -71)
+        >>> NYC = (40, -74)
+        >>> distances_mi = cityblock_distance([DC, DC], [NYC, Boston])
+        >>> np.round(distances_mi, 3).tolist()
+        [301.519, 672.089]
+
+        We can also change the units in which the distance is calculated.
+
+        >>> cityblock_distance_kilometers = CityblockDistance(unit='kilometers')
+        >>> distances_km = cityblock_distance_kilometers([DC, DC], [NYC, Boston])
+        >>> np.round(distances_km, 3).tolist()
+        [485.248, 1081.622]
+    """
+    name = "cityblock_distance"
+    input_types = [ColumnSchema(logical_type=LatLong), ColumnSchema(logical_type=LatLong)]
+    return_type = ColumnSchema(logical_type=Double, semantic_tags={'numeric'})
+    commutative = True
+
+    def __init__(self, unit='miles'):
+        if unit not in ['miles', 'kilometers']:
+            raise ValueError("Invalid unit given")
+        self.unit = unit
+
+    def get_function(self):
+        def haversine(lat_1s, lon_1s, lat_2s, lon_2s, unit):
+            lon1, lat1, lon2, lat2 = map(
+                np.radians, [lon_1s, lat_1s, lon_2s, lat_2s])
+            dlon = lon2 - lon1
+            dlat = lat2 - lat1
+            a = np.sin(dlat / 2.0) ** 2 + np.cos(lat1) * \
+                np.cos(lat2) * np.sin(dlon / 2.0)**2
+            radius_earth = 3958.7613
+            if unit == 'kilometers':
+                radius_earth = 6371.0088
+            distances = radius_earth * 2 * np.arcsin(np.sqrt(a))
+            return distances
+
+        def cityblock(latlong_1, latlong_2, unit=self.unit):
```

Comment:
Is the `unit` parameter here necessary? We should just be able to use `self.unit` within this method

Reply:
Fixed.